Source/PluginEditor.cpp

```diff
@@ -117,14 +166,91 @@ void SoomplerAudioProcessorEditor::drawSampleNameOrMessage(Graphics &g)
 
 void SoomplerAudioProcessorEditor::timerCallback()
 {
+    processor.updateTransportState();
     repaint();
 }
 
+void SoomplerAudioProcessorEditor::mouseDrag(const MouseEvent &event)
+{
+    auto position = event.getPosition();
+    if (isIntersectWithStartRangeLine(&position)) {

```

Comment:
Seems like little code duplicating. Think about moving if and else block code to some function

Reply:
There is some problems with dragging(it seems to be not so fast) so I don't want to create much of nested functions.